Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -709,6 +709,7 @@ WarpXOpenPMDPlot::DumpToFile (ParticleContainer* pc,
     }
 
     if (is_last_flush_to_step) {
+        SetupPos(currSpecies, NewParticleVectorSize, isBTD); // workaround for issue 4604
         SetConstParticleRecordsEDPIC(currSpecies, NewParticleVectorSize, charge, mass);
     }
```

Comment:
I am not sure this should go exactly like this here.

What do you think about this?
```suggestion
    if (is_last_flush_to_step && is_last_flush_and_never_particles) {
        SetupPos(currSpecies, NewParticleVectorSize, isBTD);
    }
    if (is_last_flush_to_step) {
        SetConstParticleRecordsEDPIC(currSpecies, NewParticleVectorSize, charge, mass);
    }
```

Reply:
Admittedly, this is exactly what is done a few lines earlier, isn't it? Why is that not called, I wonder.

Reply:
What are the values of `num_dump_particles` and `ParticleFlushOffset` for this problem?

Reply:
num_dump_particles=0
ParticleFlushOffset=1000

so  is_last_flush_and_never_particles (= is_last_flush_to_step && num_dump_particles == 0 && ParticleFlushOffset == 0) will be false.

Reply:
If `ParticleFlushOffset > 0` that means at some point, at least one rank hat particles.

We need to check if maybe in the call where we write particles (earlier), we have a logic mistake that forgets to call `SetupPos` collectively.

Reply:
We should printf debug for this run the `ParticleFlushOffset` and `num_dump_particles` variables. It is also possible that the book-keeping of `ParticleFlushOffset` is wrong.

Reply:
ParticleFlushOffset = 1000  (which is correct without the num_fracion option)
num_dump_particles = 0  (which is correct _with_ "num_fraction = 0" ) 

I don't know how easy it is to adjust ParticleFlushOffset with "num_fraction" on in the diags.

Reply:
Hm, maybe we filter twice? Let's take a look where `num_fraction` is applied and if the counter is done correctly afterwards.

Reply:
My understanding is that the offsets are set in the BTD object, and when it calls flush(), the  filters are then applied in the underlying flush formats during copyParticles().  Unlike other filters,  the random filter affected the offsets of the caller BTD object. 
I think the workaround is the easy and harmless. Otherwise, have to collect the particles written from all processes and update the offset in BTD.

Reply:
I would rather like to fix the underlying issue in the BTD offset counter. We need consistent meta data.

cc @RevathiJambunathan :)

Reply:
If it makes sense,  can apply the random filter before BTD calls flush().  It will make things cleaning and easier.

Reply:
Yes, we probably need to add this to:
- full diags
- BTD
- scraping